tests/jni_api.rs

```diff
@@ -143,3 +149,54 @@ pub fn with_local_frame_pending_exception() {
 
     env.exception_clear().unwrap();
 }
+
+#[test]
+pub fn call_static_method_ok() {
+    let env = attach_current_thread();
+
+    let x = JValue::from(-10);
+    let val: jint = env.call_static_method(CLASS_MATH, METHOD_MATH_ABS, SIG_MATH_ABS, &[x])
+        .expect("JNIEnv#call_static_method_unsafe should return JValue").i().unwrap();
+
+    assert_eq!(val, 10);
+}
+
+#[test]
+pub fn call_static_method_throws() {
+    let env = attach_current_thread();
+
+    let x = JValue::Long(4_000_000_000);
+    env.call_static_method(CLASS_MATH, METHOD_MATH_TO_INT, SIG_MATH_TO_INT, &[x])
+        .expect_err("JNIEnv#call_static_method_unsafe should return error");
+
+    assert!(env.exception_check().unwrap());
+    env.exception_clear().unwrap();
+}
+
+#[test]
+pub fn call_static_method_wrong_arg() {
+    let env = attach_current_thread();
+
+    let x = JValue::Double(4.56789123);
+    env.call_static_method(CLASS_MATH, METHOD_MATH_TO_INT, SIG_MATH_TO_INT, &[x])
+        .expect_err("JNIEnv#call_static_method_unsafe should return error");
+
+    assert!(env.exception_check().unwrap());
+    env.exception_clear().unwrap();
+}
+
+#[test]
+pub fn java_byte_array_from_slice() {
+    let env = attach_current_thread();
+    let buf: &[u8] = &[1, 2, 3];
+    let java_array = env.byte_array_from_slice(buf)
+        .expect("JNIEnv#byte_array_from_slice must create a java array from slice");
+    let obj = AutoLocal::new(&env, JObject::from(java_array));
+
+    assert!(!obj.as_obj().is_null());
+    let mut res: [i8; 3] = [0; 3];
+    env.get_byte_array_region(java_array, 0, &mut res).unwrap();
```

Comment:
This method implementation looks suspicios (but that may be fixed in a separate PR): won't it cause overflow of the buffer if non-zero start position is passed?

https://github.com/jni-rs/jni-rs/blob/56a14b441ca3f15e9a0db302d16b7af96bb302ee/src/wrapper/jnienv.rs#L1154-L1170

Reply:
> Most probably it will trigger the ArrayIndexOutOfBoundsException in this case.

Not necessarily — we do not check the source array length here; but `buf` overflow — most certainly.